src/containers/collectibles/solCollectibleHelpers.ts

```diff
@@ -0,0 +1,188 @@
+import {
+  Collectible,
+  CollectibleMediaType
+} from 'containers/collectibles/types'
+import {
+  SolanaNFT,
+  SolanaNFTPropertiesFile
+} from 'services/solana-client/types'
+import { Chain } from 'store/token-dashboard/slice'
+import { Nullable } from 'utils/typeUtils'
+
+import { getFrameFromGif } from './ethCollectibleHelpers'
+
+type SolanaNFTMedia = {
+  collectibleMediaType: CollectibleMediaType
+  url: string
+  frameUrl: Nullable<string>
+}
+
+const nftGif = async (nft: SolanaNFT): Promise<Nullable<SolanaNFTMedia>> => {
+  const gifFile = nft.properties.files?.find(
+    file => typeof file === 'object' && file.type === 'image/gif'
+  )
+  if (gifFile) {
+    const url = (gifFile as SolanaNFTPropertiesFile).uri
+    const frameUrl = await getFrameFromGif(url, nft.name)
+    return { collectibleMediaType: CollectibleMediaType.GIF, url, frameUrl }
+  }
+  return null
+}
+
+const nftVideo = async (nft: SolanaNFT): Promise<Nullable<SolanaNFTMedia>> => {
+  const files = nft.properties.files
+  // should we restrict video file extensions here?
+  // MP4, MOV, GLB
+  // GLTF??
+  // https://github.com/metaplex-foundation/metaplex/blob/81023eb3e52c31b605e1dcf2eb1e7425153600cd/js/packages/web/src/views/artCreate/index.tsx#L318
+  // DO WE CARE ABOUT VR NFTs??
+  const videoFile = files?.find(
+    file => typeof file === 'object' && file.type.includes('video')
+  ) as SolanaNFTPropertiesFile
+  // https://github.com/metaplex-foundation/metaplex/blob/397ceff70b3524aa0543540584c7200c79b198a0/js/packages/web/src/components/ArtContent/index.tsx#L107
+  const videoUrl = files?.find(
+    file =>
+      typeof file === 'string' &&
+      file.startsWith('https://watch.videodelivery.net/')
+  ) as string
+  const isVideo =
+    nft.properties.category === 'video' ||
+    nft.animation_url ||
+    videoFile ||
+    videoUrl
+  if (isVideo) {
+    let url: string, videoType
+    if (nft.animation_url) {
+      url = nft.animation_url
+    } else if (videoFile) {
+      url = videoFile.uri
+      videoType = videoFile.type
+    } else if (videoUrl) {
+      url = videoUrl // maybe videoUrl.replace('watch', 'iframe')?
+    } else if (files?.length) {
+      if (files.length === 1) {
+        url = typeof files[0] === 'object' ? files[0].uri : files[0]
+      } else {
+        url = typeof files[1] === 'object' ? files[1].uri : files[1]
+      }
+      // videoType = defaultVideoType
+    } else {
+      return null
+    }
+    return {
+      collectibleMediaType: CollectibleMediaType.VIDEO,
+      url,
+      frameUrl: nft.image || null
+    }
+  }
+  return null
+}
+
+const nftImage = async (nft: SolanaNFT): Promise<Nullable<SolanaNFTMedia>> => {
+  const files = nft.properties.files
+  // should we restrict image file extensions here?
```

Comment:
What are the reasons to restrict/not to restrict? Does it work with other file types out of the box? Do we not want it to? Etc. (Same with video).

I think restricting could put us in better control, say if someone tries to put a mkv or something huge on there and we have a sluggish feeling user experience. On the flip-side, the request isn't going against our server, so I'm not sure it matters? Are there security implications?

Reply:
These are all great questions. Maybe Metaplex already handles not including weird file types as they do specify what types they want [here](https://github.com/metaplex-foundation/metaplex/blob/81023eb3e52c31b605e1dcf2eb1e7425153600cd/js/packages/web/src/views/artCreate/index.tsx#L318).
For now will not restrict and we'll watch how it plays out. If need to restrict, we can base off above link